SFML/SFML
# Use std::vector.data() where appropriate

## Description

There were some places where an address of the first element of the `std::vector` was used as a pointer to array of vector's elements. These places were replaced with `std::vector.data()` that semantically better for that usage scenario.

This change was discussed in PR #1873

Two occurrences in file `examples/vulkan/Vulkan.cpp` was left untouched to avoid merge conflict with PR #1873.

Also in file `examples/vulkan/Vulkan.cpp` in `void setupRenderpass()` routine additional change to an array was made.
Array `VkAttachmentReference attachmentReferences[2];` was split into two separate variables for better readability and clearer variable usage.

I was using Windows 10 21H1 build 19043.1348 and Visual Studio 2019 (16.11.7) with Windows SDK 10.0.22000.0 to develop and test this change. 

Also I was using Ubuntu 20.04.3 with latest updates running under VirtualBox to build and test this change. Clang 10.0.0 was used for building.

## Tasks


## How to test this PR?

I've manually tested example applications to make sure that the change isn't breaking anything:
* on Windows: island, joystick, opengl, shader, sockets, sound, tennis, vulkan, win32 and window
* on Linux: island, joystick, opengl, shader, sockets, sound, tennis, window and x11

Sadly I was unable to test vulkan example on Linux. Seems that running Vulkan under VirtualBox involves some magic tricks that I'm not familiar with yet.

There are no visible changes or added features.
